staff.rb

```diff
@@ -0,0 +1,49 @@
+class Staff
+
+  attr_reader :age, :phase
+  attr_accessor :name
+
+  def initialize(options={})
+    @phase = 3
+    @age = options.fetch(:age, 0)
+    @name = options.fetch(:name, "")
+
+  end
+
+  def set_phase(num)
+    @phase = num
+    "Cool, I've always wanted to teach phase #{num}!"
+  end
+
+  def salary=(new_salary)
+    puts "This better be good!"
+    @salary = new_salary
+  end
+
+  def receive_raise(raise)
+    @salary += raise
+  end
+
+  def set_performance_rating(rating)
+
+
+    if self.class == ApprenticeTeacher
+      performance_rating = 90
+    elsif self.class == Teacher
+        performance_rating = 80
+    end
```

Comment:
use a constant instead. `PERFORMANCE_RATING_THRESHOLD = 80` at the top of the `ApprenticeTeacher` class. then the shared code can return`self.class::PERFORMANCE_RATING_THRESHOLD`. as the `performance_rating` variable.